statsmodels/tsa/vector_ar/var_model.py

```diff
@@ -802,22 +832,95 @@ def is_stable(self, verbose=False):
+            If offset is not None, then exog of the model are used if they were
+            provided in the model
+        offset : None or ndarray (steps, neqs)
+            If not None, then offset is added as an observation specific
+            intercept to the autoregression. If it is None and either trend
+            (including intercept) or exog were used in the VAR model, then
+            the linear predictor of those components will be used as offset.
+            This should have the same number of rows as steps, and the same
+            number of columns as endogenous variables (neqs).
+        seed : None or integer
+            If seed is not None, then it will be used with for the random
+            variables generated by numpy.random.
+
+        Returns
+        -------
+        endog_simulated : nd_array
+            Endog of the simulated VAR process
+
+        """
+        steps_ = None
+        if offset is None:
+            if self.k_exog_user > 0 or self.k_trend > 1:
+                # if more than intercept
+                # endog_lagged contains all regressors, trend, exog_user
+                # and lagged endog, trimmed initial observations
+                offset = self.endog_lagged[:,:self.k_exog].dot(
+                                                     self.coefs_exog.T)
+                steps_ = self.endog_lagged.shape[0]
+            else:
+                offset = self.intercept
+        else:
+            steps_ = offset.shape[0]
+
+        # default, but over written if exog or offset are used
+        if steps is None:
+            if steps_ is None:
+                steps = 1000
+            else:
+                steps = steps_
+        else:
+            if steps_ is not None and steps != steps_:
+                raise ValueError('if exog or offset are used, then steps must'
+                                 'be equal to their length or None')
+
+        y = util.varsim(self.coefs, offset, self.sigma_u, steps=steps, seed=seed)
+        return y
+
+    def plotsim(self, steps=None, offset=None, seed=None):
         """
         Plot a simulation from the VAR(p) process for the desired number of
         steps
         """
-        Y = util.varsim(self.coefs, self.exog, self.sigma_u, steps=steps)
-        plotting.plot_mts(Y)
+        y = self.simulate_var(steps=steps, offset=offset, seed=seed)
+        return plotting.plot_mts(y)
+
+    def intercept_longrun(self):
+        r"""Long run intercept of stable VAR process
+
+        Lütkepohl eq. 2.1.23
+
+        .. math:: \mu = (I - A_1 - \dots - A_p)^{-1} \alpha
+
+        where \alpha is the intercept (parameter of the constant)
+        """
+        return np.linalg.solve(self._char_mat, self.intercept)
 
     def mean(self):
-        r"""Mean of stable process
+        r"""Long run intercept of stable VAR process
+
+        Warning: trend and exog except for intercept are ignored for this.
+        This might change in future versions.
```

Comment:
Not sure how I feel about this, will give it some thought...

There seem to be a lot of methods/attributes that are conceptually (and implementation-ally) cleaner when restricting to no-trend no-exog cases.  Part of me wants to separate that (common) nice case from the VARX model/process this is turning into.

Reply:
Splitting out any of the constant-only cases will make it more difficult to add more trend/exog support later.
E.g. IRF only depends on the estimated lag polynomial, but simulated confidence intervals might require support for exog, 
currently irf_errband_mc is only for intercept case (trend in ['nc', 'c']), AFAIU.
same for FEVD if/when we get simulated or bootstrap standard errors or confidence intervals.

Reply:
> Splitting out any of the constant-only cases will make it more difficult to add more trend/exog support later

False.  Drawing the lines between classes such that they are internally coherent and represent meaningful abstractions (e.g, `VARProcess` could represent a "VAR Process") makes things easier to understand and therefore harder to screw up.

If you want to make it easier to do X later, we should be talking about improving internal consistency across the board.